### PR TITLE
Fix #139: Make RawRepresentable conform to JSONDecodable and JSONEncodable

### LINKED ISSUE #139: Enum decoding?
Is anybody is using Freddy to parse types with enums and, if so, what is your approach? I have several models that contain enums with a string raw value.

### PR DESCRIPTION
**Pull request by AquaGeek**
Fixes #139

### CODE REVIEW

File: Tests/Person.swift [hunk 1]
```diff
@@ -18,16 +25,20 @@ public struct Person: CustomStringConvertible {
...
         age = try value.int("age")
+        eyeColor = try value.decode("eyeColor")
         spouse = try value.bool("spouse")
     }
 }
 
 extension Person: JSONEncodable {
     public func toJSON() -> JSON {
-        return .Dictionary(["name": .String(name), "age": .Int(age), "spouse": .Bool(spouse)])
+        return .Dictionary(["name": .String(name), "age": .Int(age), "eyeColor": .String(eyeColor.rawValue), "spouse": .Bool(spouse)])
```
Comment (jgallagher) [inline Tests/Person.swift]: I think this could be `"eyeColor": eyeColor.toJSON()` to demonstrate that extension.

**Comment by jgallagher** [comment]
One small comment on a test, but otherwise this looks good to me. Thanks!

**Comment by AquaGeek** [comment]
Fixed

**Comment by zwaldowski** [comment]
Woot! Thank you so much!

**Comment by sprynmr** [comment]
Would love to get this in a cocoapods release. :)

**Pull request merged**